StandaloneMmPkg/Drivers/StandaloneMmCpu/EventHandle.c

```diff
@@ -36,79 +36,27 @@ MmFoundationEntryRegister (
 // event. It will either be populated in a EFI_MM_COMMUNICATE_HEADER by the
 // caller of the event (e.g. MM_COMMUNICATE SMC) or by the CPU driver
 // (e.g. during an asynchronous event). In either case, this context is
-// maintained in an array which has an entry for each CPU. The pointer to this
-// array is held in PerCpuGuidedEventContext. Memory is allocated once the
-// number of CPUs in the system are made known through the
-// MP_INFORMATION_HOB_DATA.
+// maintained in single global variable because StandaloneMm is UP-migratable
+// (which means it cannot run concurrently)
 //
-EFI_MM_COMMUNICATE_HEADER  **PerCpuGuidedEventContext = NULL;
-
-// Descriptor with whereabouts of memory used for communication with the normal world
-EFI_MMRAM_DESCRIPTOR  mNsCommBuffer;
-EFI_MMRAM_DESCRIPTOR  mSCommBuffer;
-
-MP_INFORMATION_HOB_DATA  *mMpInformationHobData;
+EFI_MM_COMMUNICATE_HEADER  *gGuidedEventContext = NULL;

```

Comment:
Honestly, I think this PR is a good opportunity to move this entire driver to ArmPkg. It is not closed related to Standalone MM, but mostly ARM specific logic.

Reply:
IIRC, RISC-V also one of the user: 
 - If you see https://edk2.groups.io/g/devel/message/108359?p=%2C%2C%2C20%2C0%2C0%2C0%3A%3Acreated%2C1%2CTuan+Phan+StandaloneMm%2C20%2C2%2C0%2C101209230, (thou there is no RISC-V platform in upstream which using StandaloneMm), it seems RISC-V uses
   this cpu driver.

So, would it better to in here??

Reply:
If RISC-V can reuse the ARM implementation for CPU driver, great! But I think if there is anything RISC-V cannot reuse, we should consider adding a RiscV package, which is outside of the scope of this PR.

Overall, I would treat this standalone package to be like MdeModulePkg which does not rely on architectural specific code.

Reply:
Alright, Then next update, I'll move StandaloneMmCoreEntryPoint && CpuDriver to ArmPkg.
Thanks!